### ISSUE TITLE
Warn when need for categorical plotting ambiguous

### ISSUE THREAD

**Issue by jlstevens** [holoviz/hvplot#941]
Issue #938 and #940 suggest categorical handling be enable when `color_key specified`
Issue #939 suggests categorical handling should be enabled for booleans e.g. `color=dim('indicator')` where this column is boolean.

However, you can also specify `cmap` with two colors, e.g. `cmap=['#FF0000', '#00FF00']` where `cmap` clearly indicates that a colorbar with smoothly varying values. What happens if `cmap` is specified but categorical handling is indicated by boolean/integer values or the presence of `color_key` as well?

I propose that a warning is issued as the user might have wanted to use `color_key` instead of `cmap` (for instance). That said, I'm not sure when discrete colorbars are appropriate versus a legend...

**Comment by jbednar** [comment]
If it's been determined to be a categorical plot, I think `color_key` should be respected, ignoring `cmap` as being almost certainly inappropriate. 

That said, if someone explicitly provides both `cmap` and `color_key`, I don't think it should be a warning, so that people can easily make an  app with a widget controlling whether it is categorical, a widget controlling the colormap, and a widget controlling the color key; we should simply use the color key if categorical and the cmap if not categorical, without complaining.

Where I think it does make sense to have a warning is when the user has has explicitly supplied a `cmap` but the plot ends up being categorical and there is no `color_key`. It seems unlikely that such a case is anything other than a user not realizing that a categorical plot needs a color key rather than a color map, so it seems like a warning is helpful there as guidance.

**Comment by jlstevens** [comment]
Right, the latter situation you describe was one of the cases I had in mind...
